src/activity_handlers.cpp

```diff
@@ -2985,7 +2985,7 @@ void activity_handlers::operation_do_turn( player_activity *act, player *p )
     const time_duration half_op_duration = difficulty * 10_minutes;
     time_duration time_left = time_duration::from_turns( act->moves_left / 100 ) ;
 
-    if( autodoc ) {
+    if( autodoc && !( p->posx() < 0 || p->posy() < 0 || p->posx() > 121 || p->posy() > 121 ) ) {
```

Comment:
If you want to check whether the position of the NPC is within the reality bubble, use this:
```suggestion
    if( autodoc && !g->m.inbounds( p->pos() ) ) {
```
Otherwise: add at least some comment why you are comparing with some magic numbers.